src/main/java/org/broadinstitute/hellbender/engine/samples/SampleDB.java

```diff
@@ -0,0 +1,317 @@
+package org.broadinstitute.hellbender.engine.samples;
+
+import htsjdk.samtools.SAMReadGroupRecord;
+import htsjdk.samtools.SAMRecord;
+import htsjdk.variant.variantcontext.Genotype;
+
+import org.broadinstitute.hellbender.exceptions.GATKException;
+import htsjdk.variant.variantcontext.Genotype;
+
+import java.util.*;
+
+/**
+ * Simple databse for managing samples
+ */
+public class SampleDB {
+    /**
+     * This is where Sample objects are stored. Samples are usually accessed by their ID, which is unique, so
+     * this is stored as a HashMap.
+     */
+    private final HashMap<String, Sample> samples = new HashMap<String, Sample>();
+
+    /**
+     * Constructor takes both a SAM header and sample files because the two must be integrated.
+     */
+    public SampleDB() {
+
+    }
+
+    /**
+     * Protected function to add a single sample to the database
+     *
+     * @param sample to be added
+     */
+    protected SampleDB addSample(Sample sample) {
+        Sample prev = samples.get(sample.getID());
+        if (prev != null) {
+            sample = Sample.mergeSamples(prev, sample);
+        }
+        samples.put(sample.getID(), sample);
+        return this;
+    }
+
+    // --------------------------------------------------------------------------------
+    //
+    // Functions for getting a sample from the DB
+    //
+    // --------------------------------------------------------------------------------
+
+    /**
+     * Get a sample by its ID
+     * If an alias is passed in, return the main sample object
+     * @param id
+     * @return sample Object with this ID, or null if this does not exist
+     */
+    public Sample getSample(String id) {
+        return samples.get(id);
+    }
+
+    /**
+     *
+     * @param read
```

Comment:
these useless tags should be made useful or removed.  Probably removed since there isn't really anything to say about them.